File: android/src/main/java/com/thermalprinterdriver/discovery/DeviceInfo.kt
package com.thermalprinterdriver.discovery

/**
 * Represents a discovered or paired Bluetooth printer device.
 */
data class DeviceInfo(
    val name: String,
    val address: String,
    val deviceType: String,  // "bt", "ble", "dual", "unknown"
    val rssi: Int = 0
)

File: ios/discovery/DeviceInfo.swift
// DeviceInfo.swift — Discovered printer descriptor
import Foundation

struct DeviceInfo {
    let name: String
    let address: String   // BLE: UUID string; LAN: "host:port"
    let deviceType: String  // "ble" | "lan" | "unknown"
    let rssi: Int?

    func toDictionary() -> [String: Any] {
        var dict: [String: Any] = [
            "name": name,
            "address": address,
            "deviceType": deviceType,
        ]
        if let rssi = rssi {
            dict["rssi"] = rssi
        }
        return dict
    }
}
